app/models/kalibro_range.rb

```diff
@@ -1,7 +1,14 @@
+require 'validators/interval_validator.rb'
+require 'validators/range_overlapping_validator.rb'
+
 class KalibroRange < ActiveRecord::Base
   belongs_to :reading
   belongs_to :metric_configuration
 
   validates :beginning, :end, :reading, :metric_configuration, presence: true
   validates :beginning, uniqueness: {scope: :metric_configuration_id, message: "Should be unique within a Metric Configuration"}
+  validates :beginning, :end, numericality: true
+  validates_with IntervalValidator, fields: [:beginning, :end], if: "beginning && self.end"
```

Comment:
Mayb comments on the purpose of those conditional validations would make them easier to understand.